src/main/scala/Switcher.scala

```diff
@@ -0,0 +1,147 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+
+class TLSplitter(n: Int, params: TLBundleParameters) extends Module {
+  val io = IO(new Bundle {
+    val in = Flipped(new TLBundle(params))
+    val out = Vec(n, new TLBundle(params))
+    val sel = Input(UInt(log2Ceil(n).W))
+  })
+
+  io.out.zipWithIndex.foreach { case (out, i) =>
+    val selected = io.sel === i.U
+    out.a.valid := io.in.a.valid && selected
+    out.a.bits := io.in.a.bits
+    out.d.ready := io.in.d.ready && selected
+
+    out.b.ready := false.B
+    out.c.valid := false.B
+    out.c.bits  := DontCare
+    out.e.valid := false.B
+    out.e.bits  := DontCare
+  }
+
+  io.in.a.ready := Vec(io.out.map(_.a.ready))(io.sel)
+  io.in.d.valid := Vec(io.out.map(_.d.valid))(io.sel)
+  io.in.d.bits  := Vec(io.out.map(_.d.bits))(io.sel)
+
+  io.in.b.valid := false.B
+  io.in.b.bits  := DontCare
+  io.in.c.ready := false.B
+  io.in.e.ready := false.B
+}
+
+class TLSwitchArbiter(n: Int, edge: TLEdge) extends Module {
+  val params = edge.bundle
+  val io = IO(new Bundle {
+    val in = Flipped(Vec(n, new TLBundle(params)))
+    val out = new TLBundle(params)
+  })
+
+  val inA = io.in.zipWithIndex.map { case (in, i) =>
+    val a = Wire(Decoupled(new TLBundleA(params)))
+    a.valid := in.a.valid
+    a.bits := in.a.bits
+    a.bits.source := Cat(in.a.bits.source, i.U(log2Ceil(n).W))
+    in.a.ready := a.ready
+    a
+  }
+
+  TLArbiter.robin(edge, io.out.a, inA:_*)
+  io.out.d.ready := false.B
```

Comment:
Can you move this to above the zip so its clear this is a different kind of default

Reply:
Ah, right